cmake/extensions.cmake

```diff
@@ -2940,6 +2943,104 @@ function(dt_chosen var)
   endif()
 endfunction()
 
+# Usage:
+#   dt_alias(<var> PROPERTY <prop>)
+#
+# Get a node path for an /aliases node property.
+#
+# Example usage:
+#
+#   # The full path to the 'led0' alias is returned in 'path'.
+#   dt_alias(path PROPERTY "led0")
+#
+#   # The variable 'path' will be left undefined for a nonexistent
+#   # alias "does-not-exist".
+#   dt_alias(path PROPERTY "does-not-exist")
+#
+# The node's path will be returned in the <var> parameter. The
+# variable will be left undefined if the alias does not exist.
+#
+# <var>           : Return variable where the node path will be stored
+# PROPERTY <prop> : The alias to check
+function(dt_alias var)
+  set(req_single_args "PROPERTY")
+  cmake_parse_arguments(DT_ALIAS "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_alias(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_ALIAS_${arg})
+      message(FATAL_ERROR "dt_alias(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_ALIAS|${DT_ALIAS_PROPERTY}")
+
+  if(${${var}} STREQUAL ${var}-NOTFOUND)
+    set(${var} PARENT_SCOPE)
+  else()
+    set(${var} ${${var}} PARENT_SCOPE)
+  endif()
+endfunction()
+
+# 4.2. *_if_dt_*
+#
+# This section is similar to the CMake generic extensions named
+# *_ifdef, except actions are performed if the devicetree
+# meets some condition.
+#
+# *_if_dt_* functions may be added as needed.
+
+# Add item(s) to a target's SOURCES list if a devicetree alias exists.
+#
+# Example usage:
+#
+#   # Adds "blink_led.c" to the sources list for the app target,
+#   # if the devicetree alias "led0" is defined.
+#   target_sources_if_dt_alias("led0" app PRIVATE blink_led.c)
+#
+# <alias>     : Devicetree alias to check
+# <target>    : Build system target whose sources to add to
+function(target_sources_if_dt_alias alias target scope item)
+  dt_alias(has_alias PROPERTY "${alias}")
+  if(DEFINED has_alias)
+    target_sources(${target} ${scope} ${item} ${ARGN})
+  endif()
+endfunction()
+
+# 4.3. *_if_no_dt_*
+#
+# This section is like the *_if_dt_* section, except actions are
+# performed if the devicetree does not meet certain criteria.
+#
+# *_if_no_dt_* functions may be added as needed.
+
+# Call message() if a devicetree alias is not present.
+#
+# The message() CMake function is called with the remaining arguments
+# if <alias> is not a devicetree alias.
+#
+# See: https://cmake.org/cmake/help/latest/command/message.html
+#
+# Example usage:
+#
+#  # Prints a fatal error and halts cmake unless a led0 devicetree
+#  # alias exists.
+#  message_if_no_dt_alias("led0" FATAL_ERROR "no LED was defined")
+#
+# <alias>     : Devicetree alias to check
+# ...         : Additional arguments are passed to message()
+function(message_if_no_dt_alias alias)
```

Comment:
This function is now used in the samples, which previously verified the `status = okay` property. Does `dt_alias` also take that property into account?